google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/project/ActiveCloudProjectHolder.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.project;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.primitives.Longs;
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Holds active cloud project, one per IDE project. Active cloud project is the last cloud project
+ * selected by a user in most of the GCP related UI reported by {@link ProjectSelector}.
+ *
+ * <p>Must only be used on Swing/IDE EDT.
+ */
+public class ActiveCloudProjectHolder {
```

Comment:
how about ActiveCloudProjectManager instead? Sounds a bit more "normal" to me than "Holder".

Reply:
Yeah, `Manager` is kind of very frequent name. It just felt this class does not really manage anything, just holds the active project per IDE project. I think `Manager` is fine as well, let me know if `Holder` seems off or not clear.

Reply:
Holder just feels strange to me. It is sort of a manager of the persistence. I'll leave it up to you.